### PR TITLE
optimize only-notify logic v2

### PR DESCRIPTION
**Pull request by klaus3000**
A new try to fix the issues (see https://github.com/PowerDNS/pdns/pull/3890). This time only the logic fixed, not the logging.

### CODE REVIEW

**Comment by rgacogne** [comment]
Travis tests failed because of a broken `jdnssec` link fixed in #4862, rebasing to master should fix that.

**Comment by klaus3000** [comment]
Rebased. Ready to accept :)

**Comment by Habbie** [comment]
Looks good to me. @mind04 any opinions?

File: docs/markdown/authoritative/settings.md [hunk 1]
```diff
@@ -585,9 +585,18 @@ This is the server ID that will be returned on an EDNS NSID query.
...
+For type=MASTER zones (or SLAVE zones with slave-renotify enabled) PowerDNS
+automatically sends NOTIFYs to the name servers specified in the NS records.
+By specifying networks/mask as whitelist, the targets can be limited. The default
+is to notify the world. To completely disable these NOTIFYs set only-notify to an
+empty value. Independent of this setting, the IP addresses or netmasks in
+[`also-notify`](#also-notify) or ALSO-NOTIFY metadata always receive AXFR NOTIFY.
+
+Note: Even if NOTIFYs are limited by a netmask, PowerDNS first has to resolve all the
+hostnames to get IP addresses. Thus, PowerDNS relies on DNS. If the respective
+authoritative name servers are slow, PowerDNS becomes slow too. To avoid this, set
```
Comment (Habbie) [inline docs/markdown/authoritative/settings.md]: Please do not write 'PowerDNS becomes slow'. Maybe write something like 'resolving all hostnames to decide who to NOTIFY may take considerable time if those names are slow to resolve'

File: docs/markdown/authoritative/settings.md [hunk 2]
```diff
@@ -585,9 +585,18 @@ This is the server ID that will be returned on an EDNS NSID query.
...
+only-notify to an empty value and specify the notification targets with ALSO-NOTIFY
+and also-notify.
```
Comment (Habbie) [inline docs/markdown/authoritative/settings.md]: and/or

**Pull request merged**